File: CalculoII/SomaDeRiemann.js
// Import the readline module
const readline = require('readline');

// Create an interface to read lines from the terminal
const rl = readline.createInterface({
	input: process.stdin,
	output: process.stdout
});

// Helper function to get user input asynchronously
function getInput(question) {
	return new Promise((resolve) => {
		rl.question(question, (answer) => {
			resolve(answer);
		});
	});
}

function SomaDeRiemann(f, a, b, n = 1000000) {

	let soma = 0;
	const deltaX = (b - a) / n;

	for (let i = 1; i <= n; i++) {
		soma += deltaX * f(a + i * deltaX);
	}

	return soma;
}

// Main function
async function main() {

	let continuar = 'yes';

	while (continuar === 'yes' || continuar === 'y') {

		const expression = await getInput("Digite a função de x (por exemplo, 'x**2', 'x*2 + 2'): ");
		const a = parseFloat(await getInput("Digite o valor de a: "));
		const b = parseFloat(await getInput("Digite o valor de b: "));

		const f = new Function("x", `return ${expression};`);

		console.log(`Soma de Riemann: ${SomaDeRiemann(f, a, b)}`);

		// Ask the user if they want to calculate another Riemann sum
		continuar = (await getInput("Deseja calcular outra soma de Riemann? (Y/N): ")).trim().toLowerCase();
		console.log();
	}

	console.log("Encerrando o programa.");
	rl.close(); // Close the readline interface
}

main();

File: Outros/CalculoII/SomaDeRiemann.py
def SomaDeRiemann(f = lambda x: 1, a = 0, b = 0, n = 1000000):
	soma = 0
	deltaX = (b - a) / n

	for i in range(1, n + 1):
		soma += deltaX * f(a + i * deltaX)

	return soma

def main():

	continuar = 'yes'

	while continuar == 'yes' or continuar  == 'y':

		expression = input("Digite a função de x (por exemplo, 'x**2', 'x*2 + 2'): ")
		a = float(input("Digite o valor de a: "))
		b = float(input("Digite o valor de b: "))

		# Converte a expressão em uma função lambda
		result = SomaDeRiemann(a = a, b = b, f = eval(f"lambda x: {expression}"))
		print(f"Soma de Riemann: {result}")

		continuar = input("Deseja calcular outra soma de Riemann? (Y/N): ").strip().lower()
		print()

	print("Encerrando o programa.")

main()
